src/menu/views/cpakfs_manager.c

```diff
@@ -0,0 +1,1016 @@
+                    sound_play_effect(SFX_EXIT);
+                } 
+                return;
+            }
+        } else if (has_mem && corrupted_pak) {
+
+            if (!show_format_controller_pak_confirm_message) {
+                if (menu->actions.back) {
+                    sound_play_effect(SFX_EXIT);
+                    menu->next_mode = MENU_MODE_BROWSER;
+                } else if (menu->actions.enter) {
+                    sound_play_effect(SFX_ENTER);
+                    show_format_controller_pak_confirm_message = true;
+                }
+            } else {
+                if (menu->actions.enter) {
+                    show_format_controller_pak_confirm_message = false;
+                    sound_play_effect(SFX_ENTER);
+                    start_format_controller_pak = true;
+                } else if (menu->actions.back) {
+                    show_format_controller_pak_confirm_message = false;
+                    sound_play_effect(SFX_EXIT);
+                } 
+            }
+        }
+    }
+}
+
+static void draw (menu_t *menu, surface_t *d) {
+    rdpq_attach(d, NULL);
+
+    ui_components_background_draw();
+
+    ui_components_layout_draw();
+
+    char has_mem_text[64];
+    char free_space_cpak_text[64];
+    menu_font_type_t style;
+
+    style = STL_DEFAULT;
+
+    if (has_mem) {
+        sprintf(has_mem_text, "CPAK detected");
+        style = STL_GREEN;
+
+        if (has_mem && !corrupted_pak) {
+            style = STL_GREEN;
+            sprintf(free_space_cpak_text, "%d/123 free blocks", cpakfs_stats.pages.total - cpakfs_stats.pages.used);
+        } else if (has_mem && corrupted_pak) {
+            sprintf(has_mem_text, "%s %s", has_mem_text, " (is NOT valid. Corrupted)");
+            style = STL_ORANGE;
+            sprintf(free_space_cpak_text, " ");
+        }
+    } else {
+        sprintf(has_mem_text, "NO CPAK detected");
+        style = STL_ORANGE;
+        sprintf(free_space_cpak_text, " ");
+    }
+
+    ui_components_main_text_draw(STL_DEFAULT,
+        ALIGN_CENTER, VALIGN_TOP,
+        "CONTROLLER PAK MANAGEMENT\n"
+    );
+
+    ui_components_main_text_draw(STL_DEFAULT,
+        ALIGN_RIGHT, VALIGN_TOP,
+        "B: Back\n"
+    );
+
+    ui_components_main_text_draw(STL_DEFAULT,
+        ALIGN_LEFT, VALIGN_TOP,
+        "\n"
+        "Controller: < %d >\n",
+            controller_selected + 1
+    );
+
+    ui_components_main_text_draw(style,
+        ALIGN_LEFT, VALIGN_TOP,
+        "\n"
+        "                   %s\n",
+        has_mem_text
+    );
+
+    if (has_mem) {
+        ui_components_main_text_draw(STL_DEFAULT,
+            ALIGN_LEFT, VALIGN_TOP,
+            "\n"
+            "                                  %s\n",
+            free_space_cpak_text
+        );
+
+        ui_components_main_text_draw(STL_DEFAULT,
+            ALIGN_LEFT, VALIGN_TOP,
+            "\n"
+            "\n"
+            "            Name           Code    Ext.        Size\n"
```

Comment:
```suggestion
            "            Name           Code    Ext.        Size [blocks]\n"
```
Then you dont have to show it in every entry?